packages/dx-react-grid/docs/reference/table-tree-column.md

```diff
@@ -0,0 +1,112 @@
+- [SelectionState](selection-state.md) [Optional]
+- [IntegratedSelection](integrated-selection.md) [Optional]
+- [TableHeaderRow](table-header-row.md) [Optional]
+
+### Properties
+
+Name | Type | Default | Description
+-----|------|---------|------------
+for | string | | A column name used to identify a column that will be represented as a tree.
+cellComponent | ComponentType&lt;[TableTreeColumn.CellProps](#tabletreecolumncellprops)&gt; | | A component that renders a cell within a data row.
+indentComponent | ComponentType&lt;[TableTreeColumn.IndentProps](#tabletreecolumnindentprops)&gt; | | A component that renders an indent used to identify row level.
+toggleButtonComponent | ComponentType&lt;[TableTreeColumn.ToggleButtonProps](#tabletreecolumntogglebuttonprops)&gt; | | A component that renders a button that controls row's expanded state.
+checkboxComponent | ComponentType&lt;[TableTreeColumn.CheckboxProps](#tabletreecolumncheckboxprops)&gt; | | A component that renders a checkbox used to control selection.
+showSelectionControls? | boolean | false | Specifies whether to render selection controls. Requires the [SelectionState](selection-state.md) dependency.
+showSelectAll? | boolean | false | Specifies whether to render Select All checkbox. Requires the [IntegratedSelection](integrated-selection.md) dependency.
+
+## Interfaces
+
+### TableTreeColumn.CellProps
+
+Describes properties passed to a component that renders a cell within a data row.
+
+Extends [Table.CellProps](table.md#tablecellprops)
+
+Field | Type | Description
+------|------|------------
+value | any | Specifies a value to be rendered within the cell.
+row | any | Specifies the cell's row.
+column | [Column](grid.md#column) | Specifies the cell's column.
+controls? | ReactNode | A React node to be placed in the cell part that display controls.
+children? | ReactNode | A React node to be placed in the cell part that display cell value.
+
+### TableTreeColumn.IndentProps
+
+Describes properties passed to a component that renders an indent used to identify row level.
+
+Field | Type | Description
+------|------|------------
+level | number | Specifies a row level.
+
+### TableTreeColumn.ToggleButtonProps
+
+Describes properties passed to a component that renders a button that controls row's expanded state.
+
+Field | Type | Description
+------|------|------------
+visible | boolean | Specifies whether to show button.
+expanded | boolean | Specifies whether a row is expanded.
+onToggle | () => void | An event that initiates row expanding or collapsing.
+
+### TableTreeColumn.CheckboxProps
+
+Describes properties passed to a component that renders a checkbox used to control selection.
+
+Field | Type | Description
+------|------|------------
+disabled | boolean | Specifies whether a row is disallowed to select.
+selected | boolean | Specifies whether a row is selected.
+someSelected | boolean | Specifies whether a row is partially selected.
```

Comment:
It seems strange for a checkbox to have such properties. `checked`, `indeterminate` and `onChange` would be more natural.